pkg/pipeline/pipeline_test.go

```diff
@@ -102,6 +91,54 @@ func Test_SimplePipeline(t *testing.T) {
 	ingester := mainPipeline.Ingester.(*ingest.IngestFile)
 	decoder := mainPipeline.Decoder.(*decode.DecodeJson)
 	writer := mainPipeline.Writer.(*write.WriteNone)
+	require.Equal(t, 5103, len(ingester.PrevRecords))
 	require.Equal(t, len(ingester.PrevRecords), len(decoder.PrevRecords))
 	require.Equal(t, len(ingester.PrevRecords), len(writer.PrevRecords))
+
+	// checking that the processing is done for at least the first line of the logs
+	require.Equal(t, ingester.PrevRecords[0], decoder.PrevRecords[0])
+	// values checked from the first line of the ../../hack/examples/ocp-ipfix-flowlogs.json file
+	require.Equal(t, config.GenericMap{
+		"fl2m_bytes":   float64(20800),
+		"fl2m_dstAddr": "10.130.2.2",
+		"fl2m_dstPort": float64(36936),
+		"fl2m_packets": float64(400),
+		"fl2m_srcAddr": "10.130.2.13",
+		"fl2m_srcPort": float64(3100),
+	}, writer.PrevRecords[0])
+}
+
+func BenchmarkPipeline(b *testing.B) {
+	logrus.StandardLogger().SetLevel(logrus.ErrorLevel)
+	t := &testing.T{}
+	loadGlobalConfig(t)
+	config.Opt.PipeLine.Ingest.Type = "file_chunks"
+	if t.Failed() {
+		b.Fatalf("unexpected error loading config")
+	}
+	for n := 0; n < b.N; n++ {
+		b.StopTimer()
+		p, err := NewPipeline()
+		if err != nil {
+			t.Fatalf("unexpected error %s", err)
+		}
+		b.StartTimer()
```

Comment:
Where is the timing information used?

Reply:
`StopTimer` and `StartTimer` are used to exclude from the benchmark some parts of the code that you don't want to measure. In this case we don't want to measure the Pipeline creation time but just the amount of metrics that you can forward.